tests/unit/artifact/test_repository.py

```diff
@@ -313,3 +317,190 @@ def test_parse_rpm_string_valid(pkg_string, expected):
 def test_parse_rpm_string_invalid(pkg_string):
     with pytest.raises(ValueError, match=r"does not match|Malformed package string"):
         parse_rpm_string(pkg_string)
+
+
+# ================================================================================
+# Tests for RepositoryFileProvider
+# ================================================================================
+
+
+def test_repository_provider_id_extraction(root_logger):
+    """Test provider ID extraction from raw provider ID"""
+
+    # Valid provider ID
+    raw_id = "repository-url:https://download.docker.com/linux/centos/docker-ce.repo"
+    provider = RepositoryFileProvider(raw_id, root_logger)
+    assert provider.id == "https://download.docker.com/linux/centos/docker-ce.repo"
+
+
+def test_repository_provider_invalid_format(root_logger):
+    """Test that invalid provider ID formats raise ValueError"""
+
+    # Missing prefix
+    with pytest.raises(ValueError, match="Invalid repository provider format"):
+        RepositoryFileProvider("https://example.com/repo.repo", root_logger)
+
+    # Empty URL after prefix
+    with pytest.raises(ValueError, match="Missing repository URL"):
+        RepositoryFileProvider("repository-url:", root_logger)
+
+
+def test_repository_provider_artifacts_before_fetch(root_logger):
+    """Test that accessing artifacts before fetch_contents raises error"""
+
+    provider = RepositoryFileProvider("repository-url:https://example.com/test.repo", root_logger)
+
+    with pytest.raises(GeneralError, match="Call fetch_contents first"):
+        _ = provider.artifacts
+
+
+@patch('tmt.steps.prepare.artifact.providers.tmt.utils.retry_session')
+def test_repository_provider_fetch_contents(mock_retry_session, root_logger):
+    """Test fetch_contents method discovers RPMs from repository"""
+
+    # Mock the Repository.from_url call
+    mock_session = MagicMock()
+    mock_response = MagicMock()
+    mock_response.text = VALID_REPO_CONTENT
+    mock_response.raise_for_status.return_value = None
+    mock_session.get.return_value = mock_response
+    mock_retry_session.return_value.__enter__.return_value = mock_session
+
+    # Mock guest and package manager
+    mock_guest = MagicMock()
+    mock_package_manager = MagicMock()
+    mock_guest.package_manager = mock_package_manager
+
+    # Mock list_packages to return some RPMs
+    mock_package_manager.list_packages.return_value = [
+        "docker-ce-1:20.10.7-3.el8.x86_64",
+        "docker-ce-cli-1:20.10.7-3.el8.x86_64",
+        "containerd.io-1.4.6-3.1.el8.x86_64",
+    ]
+
+    provider = RepositoryFileProvider(
+        "repository-url:https://download.docker.com/linux/centos/docker-ce.repo", root_logger
+    )
+
+    # Call fetch_contents
+    result = provider.fetch_contents(mock_guest, Path("/tmp/artifacts"))
```

Comment:
Please use a `tmp_path` instead of a hard-coded path. We should not create any paths that are not managed by pytest.